code401-challenges/src/main/java/code401/challenges/linkedlist/LinkedList.java

```diff
@@ -135,13 +135,51 @@ public int insertAfter(int value, int newVal){
                     newValue.next = current.next;
                     current.next = newValue;
                     this.size++;
-                    return 0;
+                    return;
                 }else{
                     current = current.next;
                 }
             }
         }
         System.out.println("The linked list does not contain "+ value);
+        return;
+    }
+
+    public int knthFromEnd(int value){
+        if(this.head == null){
+            throw new NullPointerException("The Linked list is empty");
```

Comment:
You don't always want to throw errors! It's perfectly reasonable to pass in an empty LinkedList into this function or pass a value that is larger than the length of the LinkedList. Instead of throwing an error (which will cause the code to stop) it would be better to handle the issue so your code will continue to run